src/common/components/recovery-account/index.tsx

```diff
@@ -0,0 +1,325 @@
+import { Button, Form, FormControl, Modal } from "react-bootstrap";
+
+import { ActiveUser } from "../../store/active-user/types";
+import { Global } from "../../store/global/types";
+import { PrivateKey } from "@hiveio/dhive";
+
+import keyOrHot from "../key-or-hot";
+import UserAvatar from "../user-avatar";
+import { error } from "../feedback";
+import PopoverConfirm from "../popover-confirm";
+import LinearProgress from "../linear-progress";
+
+import { arrowRightSvg } from "../../img/svg";
+import { _t } from "../../i18n";
+import { getAccount, findAccountRecoveryRequest } from "../../api/hive";
+import {
+  changeRecoveryAccount,
+  formatError,
+  changeRecoveryAccountHot,
+  changeRecoveryAccountKc
+} from "../../api/operations";
+
+interface Props {
+  global: Global;
+  activeUser: ActiveUser | null;
+  signingKey: string;
+  setSigningKey: (key: string) => void;
+}
+export default function AccountRecovery(props: Props) {
+  const [keyDialog, setKeyDialog] = useState(false);
+  const [inProgress, setInProgress] = useState(false);
+  const [disabled, setDisabled] = useState(true);
+  const [popOver, setPopOver] = useState(false);
+  const [step, setStep] = useState(1);
+  const [toError, setToError] = useState("");
+  const [currRecoveryAccount, setCurrRecoveryAccount] = useState("");
+  const [newRecoveryAccount, setNewCurrRecoveryAccount] = useState("");
+
+  useEffect(() => {
+    getCurrentAccount();
+    getRecoveryRequest();
+  }, []);
+
+  const getCurrentAccount = async () => {
+    const account = await getAccount(props.activeUser!.username);
+    const { recovery_account } = account;
+    setCurrRecoveryAccount(recovery_account);
+  };
+
+  const getRecoveryRequest = async () => {
+    const resp = await findAccountRecoveryRequest(props.activeUser!.username);
+    if (resp.requests.length) {
+      setPopOver(true);
+    }
+  };
+
+  const toggleKeyDialog = () => {
+    setKeyDialog(!keyDialog);
+  };
+
+  const newRecoveryAccountChange = async (
+    e: React.ChangeEvent<typeof FormControl & HTMLInputElement>
+  ) => {
+    e.persist();
+    setNewCurrRecoveryAccount(e.target.value);
+
+    if (e.target.value.length == 0) {
+      setDisabled(true);
+      setToError("");
+      return;
+    } else {
+      const resp = await getAccount(e.target.value);
+      if (resp) {
+        setDisabled(false);
+        setToError("");
+      } else {
+        if (e.target.value.length > 0) {
+          setDisabled(true);
+          setToError(_t("account-recovery.to-not-found"));
+        }
+      }
+    }
+  };
+
+  const update = () => {
+    if (!popOver) {
+      setKeyDialog(true);
+    }
+  };
+
+  const onKey = (key: PrivateKey): void => {
+    setInProgress(true);
+    let promise: Promise<any> = changeRecoveryAccount(
+      props.activeUser!.username,
+      newRecoveryAccount,
+      [],
+      key
+    );
+    promise
+      .then((resp) => {
+        if (resp.id) {
+          setInProgress(false);
```

Comment:
It should be false in `finally` not here

Reply:
@dkildar Updated.